File: app/ApplicationConfig.scala
import java.util.UUID

//import models.base.AbstractEntity
//import models.content.ContentPage
//import org.neo4j.graphdb.Node
//import org.springframework.context.annotation.{Bean, Configuration}
//import org.springframework.context.ApplicationListener
//import org.springframework.data.neo4j.config.{Neo4jConfiguration, EnableNeo4jRepositories}
//import org.springframework.data.neo4j.lifecycle.BeforeSaveEvent

//@Configuration
//@EnableNeo4jRepositories
//class ApplicationConfig extends Neo4jConfiguration {

//  @Bean
//  def beforeSaveEventApplicationListener(): ApplicationListener[BeforeSaveEvent[Node]] = {
//    return new ApplicationListener[BeforeSaveEvent[Node]]() {
//
//      @Override
//      override def onApplicationEvent(event: BeforeSaveEvent[Node]) {
//        val entity = event.getEntity().asInstanceOf[AbstractEntity]
//        val graphid = entity
//      }
//    }
//  }
//
//  @Bean
//  def beforeSaveEventApplicationListener(): ApplicationListener[BeforeSaveEvent[ContentPage]] = {
//    new ApplicationListener[BeforeSaveEvent[ContentPage]]() {
//
//      override def onApplicationEvent(event: BeforeSaveEvent[ContentPage]) {
//        val entity = event.getEntity
//        //entity.setUniqueId(UUID.randomUUID)
//      }
//    }
//  }
//
//  @Bean
//  def afterSaveEventApplicationListener(): ApplicationListener[AfterSaveEvent] = {
//    new ApplicationListener[AfterSaveEvent]() {
//
//      override def onApplicationEvent(event: AfterSaveEvent) {
//        val entity = event.getEntity.asInstanceOf[AbstractEntity]
//        Logger.debug("Saved entity (" + entity.id + ") objectId(" + entity.objectId + ")")
//      }
//    }
//  }
//
//  @Bean
//  def deleteEventApplicationListener(): ApplicationListener[AfterDeleteEvent] = {
//    new ApplicationListener[AfterDeleteEvent]() {
//
//      override def onApplicationEvent(event: AfterDeleteEvent) {
//        Logger.debug("Deleted entity")
//      }
//    }
//  }
//}

File: app/ApplicationConfig.java
/*
import org.neo4j.ogm.session.Session;
import org.neo4j.ogm.session.SessionFactory;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Scope;
import org.springframework.context.annotation.ScopedProxyMode;
import org.springframework.data.neo4j.config.Neo4jConfiguration;
import org.springframework.data.neo4j.repository.config.EnableNeo4jRepositories;
import org.springframework.data.neo4j.server.Neo4jServer;
import org.springframework.data.neo4j.server.RemoteServer;
import org.springframework.transaction.annotation.EnableTransactionManagement;

@Configuration
@EnableNeo4jRepositories("repository")
@EnableTransactionManagement
@ComponentScan({"models", "controllers", "services", "repositories"})
public class ApplicationConfig extends Neo4jConfiguration {

    public static final int NEO4J_PORT = 7474;

    @Override
    public SessionFactory getSessionFactory() {
        return new SessionFactory("_root_.models");
    }

    @Bean
    public Neo4jServer neo4jServer() {
        return new RemoteServer("http://localhost:" + NEO4J_PORT);
    }

    @Override
    @Bean
    @Scope(value = "session", proxyMode = ScopedProxyMode.TARGET_CLASS)
    public Session getSession() throws Exception {
        return super.getSession();
    }
}
*/